tests/unit/server.test.js

```diff
@@ -81,6 +81,74 @@ test.cb('that calling close will trigger the onclose of websockets', t => {
   };
 });
 
+test.cb('that calling close on the server without a `code` will result in a default code of `1000`', t => {
+  const myServer = new Server('ws://not-real/');
+
+  myServer.on('connection', () => {
+    myServer.close({
+      reason: 'Some reason'
+    });
+  });
+
+  const client = new WebSocket('ws://not-real/');
+  client.onclose = event => {
+    t.is(event.code, 1000, 'the default code `1000` was recieved');
+    t.end();
+  };
+});
+
+// https://github.com/thoov/mock-socket/pull/391
+test.cb('that calling close on the server with `code` set to `0` will result in a code of `0`', t => {
```

Comment:
the expecting behaviour is `calling close with a code that is not a number, < 1000, or > 4999 throws an error`, so:

- socket.close() - ok, code === 1000
- socket.close(0) - fail
- socket.close(null) - fail

Reply:
This library is not an _implementation_ of a websocket server, but rather a _mock_ of one. My expectation of a mock is that I can configure it to behave in a certain way, pass it to another system, and have it embody that behaviour using an interface that conforms to the interface the system expects.

Said another way, if I configure the mock to close the socket with error code `0`, I expect it to fire the close event with error code `0`. That is the aim of this PR.

Reply:
It’s what I’m really trying to understand. Is it a possible case closing the connection on the server with the `0` code? As currently I see it’s not supported by design (I sent you an rfc link).

Another words, this library can be used to mock a *real* server with the *real* behaviour. If the case is not possible at all, so this library should say about this.

So, if you have such case, please send a link to part of the code of the *real* server. I would really like to sort this out.

Reply:
I think we'll have to agree to disagree about the role of this mock. I contend that the mock should behave how I tell it to, and you believe that the mock should be opinionated and reject certain configurations.

We don't need to agree, we only need to resolve the following:

What would you write here in order to make the mock throw an error having `wasClean` equal to false.

```ts
ws.close({ /* what would you write here? */ });
```

Reply:
Here're the difference.

```ts
test("closes normally", (done) => {
  const server = new Server("ws://test");
  const ws = new WebSocket("ws://test");

  server.on("connection", (client) => {
    client.close(); // the same as client.close({ code: 1000 });
  });

  ws.onclose = (e) => {
    expect(e.code).toBe(1000);
    expect(e.wasClean).toBe(true);

    done();
  };
});

test("closes wasClean=false", (done) => {
  const server = new Server("ws://test");
  const ws = new WebSocket("ws://test");

  server.on("connection", (client) => {
    client.close({ code: 1001 });
  });

  ws.onclose = (e) => {
    expect(e.code).toBe(1001);
    expect(e.wasClean).toBe(false);

    done();
  };
});
```

According to rfc, `e.code` can not be less than 1000. But how is your issue related to `wasClean`?